api/helpers/utils.js

```diff
@@ -223,19 +222,11 @@ exports.runDataQuery = async function (modelType, role, userGuid, query, fields,
                     }
                   ]
                 },
-                // Check if user either has the create-projects role or has project permissions.
-                { $cond: 
-                  { if: { $in: [ "public", role ] }, then: true, else:
-                    { $cond: 
-                      { if: isUserQuery, then: true, else: 
```

Comment:
This is where the exception for users would normally happen. Typically schema visibility for authenticated calls (without the "public" role) would need to include either the "create-projects" role or that the user has been added to the project permissions